### PR TITLE
createsUnitsList can be zero

### PR DESCRIPTION
**Pull request by ron-murhammer**
Address https://forums.triplea-game.org/topic/1945/trigger-createsunitslist-count-from-1-to-0

### CODE REVIEW

**Comment by ron-murhammer** [comment]
Merging as very minor change that can be tested by the map maker.

**Comment by FrostionAAA** [comment]
This change seems to work. I don’t know if it is intentionally, but I would have imagined that the generated tooltip info would not state the fact that a unit does not produce another unit anymore. I would have expected the text “1xBattle-Chicken” to just disappear, not to become “0xBattle-Chicken”

example

I guess it is a consequence of the unit actually still having the unit attachment <option name="createsUnitsList" value="0: Battle-Chicken"/>. But if I am not mistaken, not all unit options with value 0 or false are shown? Right? So it could also be possible to hide this one?

File: game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java [hunk 1]
```diff
@@ -1920,8 +1920,8 @@ private void setCreatesUnitsList(final String value) throws GameParseException {
           "createsUnitsList: No unit called:" + unitTypeToProduce + thisErrorMsg());
     }
     final int n = getInt(s[0]);
-    if (n < 1) {
-      throw new GameParseException("createsUnitsList must have positive values" + thisErrorMsg());
+    if (n < 0) {
+      throw new GameParseException("createsUnitsList cannot have negative values" + thisErrorMsg());
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java:1924]: nit, missing the trailing space before 'thisErrorMsg()'

suggestion: we might have a better error message if we print out the `s` array to give a person more context to where the error is, particularly if there are many create units lists.

**Comment by DanVanAtta** [comment]
@ron-murhammer the comment on which feature is being addressed is useful, but IMO it would be more useful in the commit history. If there is no review, then any PR commentary is of zero value - the PR overview and comments are for the reviewer.

**Comment by DanVanAtta** [comment]
An even better commit comment would have stated that this change was to support examples such as:
```
<option name="unitProperty" value="createsUnitsList" count="0:Battle-Chicken"/>
```

IMO we should decide if we want our commit history to be a useful tool or not to bother trying -  that discussion is in: https://github.com/triplea-game/triplea/issues/6316

**Pull request merged**